sycl/plugins/esimd_emulator/pi_esimd_emulator.cpp

```diff
@@ -909,6 +874,30 @@ pi_result piContextRelease(pi_context Context) {
   return PI_SUCCESS;
 }
 
+bool _pi_context::checkSurfaceArgument(pi_mem_flags Flags, void *HostPtr) {
+  if (Flags & (PI_MEM_FLAGS_HOST_PTR_USE | PI_MEM_FLAGS_HOST_PTR_COPY)) {
+    if (HostPtr == nullptr) {
+      if (PrintPiTrace) {
+        std::cerr << "HostPtr argument is required for "
```

Comment:
for future: it would be good to define error function which provides component id (i.e. "ESIMD emulator") in error messages, so that user knows where it comes from